codex/blockexchange/network/network.nim

```diff
@@ -244,7 +246,8 @@ proc rpcHandler(
 
     if payment =? SignedState.init(msg.payment):
       asyncSpawn b.handlePayment(peer, payment)
-
+  except CancelledError as error:
```

Comment:
If we're going to `asyncSpawn handlePayment`, there is no point in `try/catch` around it. Any escaping exception is converted to a `Defect` by `asyncSpawn`.

Reply:
Good point, I have no idea why the error handling was here in the first place. I just added the extra case for `CancelledError`, but in the current code everything is `asyncSpawn`ed, and no errors will be raised anyway.